src/main/java/org/wildfly/security/credential/X509CertificateChainPrivateCredential.java

```diff
@@ -47,9 +47,15 @@ public X509CertificateChainPrivateCredential(final PrivateKey privateKey, final
         if (! getFirstCertificate().getPublicKey().getAlgorithm().equals(privateKey.getAlgorithm())) {
             throw ElytronMessages.log.mismatchedPublicPrivateKeyAlgorithms();
         }
-        if (! KeyUtil.hasSameParameters(getFirstCertificate().getPublicKey(), privateKey)) {
-            throw ElytronMessages.log.mismatchedPublicPrivateKeyParameters();
-        }
+
+        /*
+         * Where keys from PKCS#11 are in use we can not guarantee to convert them to a comparable parameter spec
+         * which leads to the check below returning false - the most we can check is matching algorithms which are checked above.
+         */
+
+        //if (! KeyUtil.hasSameParameters(getFirstCertificate().getPublicKey(), privateKey)) {
+        //    throw ElytronMessages.log.mismatchedPublicPrivateKeyParameters();
+        //}
         this.privateKey = privateKey;
```

Comment:
Looking at this `privateKey` handling I wonder if it is available while using PKCS11 storage.

[1] https://issues.jboss.org/browse/JBEAP-10396?focusedCommentId=13395133&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13395133